### ISSUE TITLE
Multiple annotations should be allowed on the same line when construct is a constructor

### ISSUE THREAD

**Issue by ZacSweers** [pinterest/ktlint#673]
Consider the following example:

```kotlin
class Foo @JvmOverloads @SuppressLint("CheckResult") constructor(...)
```

There alternative is basically this

```kotlin
class Foo
@JvmOverloads
@SuppressLint("CheckResult")
constructor(...)
```

Which I'm not sure is better. From `experimental:annotation`

**Comment by Tapchicoma** [comment]
I think in this case behaviour of `experimental:annotation` is correct. According to https://kotlinlang.org/docs/reference/coding-conventions.html#annotation-formatting:
> A single annotation without arguments may be placed on the same line as the corresponding declaration:

Another problem having multiple annotations for constructor on the same line - this line could more easily hit max-line limit, especially with long class name.

**Comment by ZacSweers** [comment]
I think it makes sense for the declaration examples they have there, but not constructors. Yes it increases line length, but unlike the examples they have there, new lines actually have a destructive effect on the class header declaration here.

**Comment by 3flex** [comment]
Came here to report the same... I think without clear examples on the Kotlin coding conventions page for constructor annotations, this isn't something that should be flagged by the rule.

**Comment by lwasyl** [comment]
I'd like to expand the discussion to not only constructor annotations, but annotations for method parameters too. Currently (0.44.0) this is reported as an error:

```kotlin

@Component(modules = [Foo::class, Foo::class])
interface Foo {

    @Component.Factory
    interface Factory {

        fun create(
            @BindsInstance context: Context,
            @BindsInstance @Bar bar: String,
        ): Foo
    }
}

annotation class Bar
```

and I'm not sure what would be the suggested formatting. The following is the only variant that KtLint doesn't complain about, and I argue it's much less readable as long as the annotations fit on a single line (and they mostly do)

```kotlin
@Component(modules = [Foo::class, Foo::class])
interface Foo {

    @Component.Factory
    interface Factory {

        fun create(
            @BindsInstance context: Context,
            @BindsInstance
            @Bar
            bar: String,
        ): Foo
    }
}

annotation class Bar
```

**Comment by MadeInChina** [comment]
I have same issue with - Multiple annotations should not be placed on the same line as the annotated construct (annotation)
E.g
`    fun update(@Valid @Argument("input") input: UpdateRequest): UpdateResponse {}`

**Comment by hantsy** [comment]
How to disable this rule?

**Comment by paul-dingemans** [comment]
> ```kotlin
> fun update(@Valid @Argument("input") input: UpdateRequest): UpdateResponse {}
> ```

Please read the documentation: https://pinterest.github.io/ktlint/faq/#how-do-i-suppress-errors-for-a-lineblockfile

**Comment by paul-dingemans** [comment]
Behavior of the `annotation` rules has been made more consistent in PR's #1916 and #1917 albeit not in the way most commentors in this issue would like it to be changed. For a consistency viewpoint, annotation wrapping should be done as consistent/deterministic as possible. Unfortunately, it is not possible to create a formatter which makes everybody happy without having hundreds of parameters to tweak the behavior.
